lib/src/test/kotlin/starknet/data/TypedDataTest.kt

```diff
@@ -704,6 +706,73 @@ internal class TypedDataTest {
         }
     }
 
+    @Nested
+    inner class JsonConversionTest {
+        private val types = mapOf(
+            "StarknetDomain" to listOf(
+                TypedData.StandardType("name", "shortstring"),
+                TypedData.StandardType("version", "shortstring"),
+                TypedData.StandardType("chainId", "shortstring"),
+                TypedData.StandardType("revision", "shortstring"),
+            ),
+            "Example" to listOf(
+                TypedData.StandardType("n0", "felt"),
+                TypedData.StandardType("n1", "bool"),
+                TypedData.StandardType("n2", "string"),
+                TypedData.StandardType("n3", "selector"),
+                TypedData.StandardType("n4", "u128"),
+                TypedData.StandardType("n5", "i128"),
+                TypedData.StandardType("n6", "ContractAddress"),
+                TypedData.StandardType("n7", "ClassHash"),
+                TypedData.StandardType("n8", "timestamp"),
+                TypedData.StandardType("n9", "shortstring"),
+            ),
+        )
+        private val domainObject = """
+            {
+                "name": "StarkNet Mail",
+                "version": "1",
+                "chainId": "1",
+                "revision": 1
+            }
+        """.trimIndent()
+        private val messageObject = """
+            {
+                "n0": "0x3e8",
+                "n1": true,
+                "n2": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+                "n3": "transfer",
+                "n4": "0x3e8",
+                "n5": "-170141183460469231731687303715884105727",
+                "n6": "0x3e8",
+                "n7": "0x3e8",
+                "n8": 1000,
+                "n9": "transfer"
+            }
+        """
+        private val td = TypedData(
+            types = types,
+            primaryType = "Example",
+            domain = domainObject,
+            message = messageObject,
+        )
+        private val tdJsonString by lazy { File("$TYPE_DATA_DIR_PATH/rev_1/typed_data_basic_types_example.json").readText() }
+
+        @Test
+        fun `typed data toJsonString`() {
+            // TypedData.toJsonString() returns a JSON string without indentations
+            // So we need to compare it with the expected JSON string without indentations
+            val expectedJsonString = Json.encodeToString(Json.parseToJsonElement(tdJsonString))
```

Comment:
`TypedData.toJsonString()` returns a minified JSON string, so we need to convert pretty-printed `tdJsonString` to minified format.